src/librustc/middle/traits/object_safety.rs

```diff
@@ -104,9 +110,6 @@ fn object_safety_violations_for_trait<'tcx>(tcx: &ty::ctxt<'tcx>,
         .collect();
 
     // Check the trait itself.
-    if trait_has_sized_self(tcx, trait_def_id) {
```

Comment:
No, I also thought about that, but it doesn't seem to work. This is the `violations` vector inside `object_safety_violations_for_trait`. This one will only ever have one `SizedSelf` in it.